Algorithm/QCAlgorithm.History.cs

```diff
@@ -629,8 +629,8 @@ private IEnumerable<HistoryRequest> CreateBarCountHistoryRequests(IEnumerable<Sy
             {
                 var res = GetResolution(x, resolution);
                 var exchange = GetExchangeHours(x);
-                var start = _historyRequestFactory.GetStartTimeAlgoTz(x, periods, res, exchange);
-                var end = Time.RoundDown(res.ToTimeSpan());
+                var start = _historyRequestFactory.GetStartTimeAlgoTz(x, periods, res, exchange, GetDataTimeZone(x));
```

Comment:
If we got the matching subscriptions before I think we could also use `config.DataTimeZone`?